modules/mcuboot/hooks/CMakeLists.txt

```diff
@@ -0,0 +1,16 @@
+#
+# Copyright (c) 2022 Nordic Semiconductor
+#
+# SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+#
+
+if(CONFIG_BOOT_IMAGE_ACCESS_HOOKS)
+  zephyr_interface_library_named(NRF53HOOKS)
```

Comment:
why do you want an interface library here ?

Who is expected to link to `NRF53HOOKS` (except the zephyr lib in here containing `nrf53_hooks.c`) ?